src/analyzer/analyzerwaveform.cpp

```diff
@@ -75,6 +82,11 @@ bool AnalyzerWaveform::initialize(TrackPointer tio, int sampleRate, int totalSam
         m_waveformSummary = WaveformPointer(new Waveform(
                 sampleRate, totalSamples, mainWaveformSampleRate,
                 summaryWaveformSamples));
+        //I force it to NOT dirty, because other Mixxx threads like the Main thread can trigger 
+        //TrackDAO.updateTrack(), which would try to save an empty or incomplete analysis. 
+        m_waveform->setDirty(false);
```

Comment:
The waveform has just been created (new Waveform()) . They haven't even been set to the track (That's done just after the setDirty).

The issue is already happening in master. It is unrelated to my multithreaded change. That's why i am spliting the change.

As an explanation, what I saw in the log file was that the analysis scan started and the waveform was marked dirty (I could see that the analyzer thread tries to locate an previous analysis which is still non-existant). It continues scanning and then, the Main thread calls the analyzerdao to save (this is triggered by the cachereader, i think, but i cannot assure it).
Later on, I see the analyzer ending the scan, and since it detects that the waveform is "clean" (because it has been set to clean by that previous save), then it does not save to the database.

The waveform needs to be set to the track, because else, Mixxx would not redraw the waveform in realtime while it is being generated, but if it is dirty before it has finished, a track update process can try to save it at any time and marking it "clean".

I thought that the most logical thing, (given that it is only set to dirty on the analyzer), is to mark it dirty only when we want it to get saved.